central/image/service/service_impl.go

```diff
@@ -485,6 +489,23 @@ func (s *serviceImpl) EnrichLocalImageInternal(ctx context.Context, request *v1.
 	return internalScanRespFromImage(img), nil
 }
 
+// buildNames returns a slice containing the known image names from the various parameters.
+func buildNames(srcImage *storage.ImageName, metadata *storage.ImageMetadata) []*storage.ImageName {
```

Comment:
I don't believe so. Only delegated scanning will support mirroring - which means as it stands only the workflow from Sensor to `EnrichLocalImageInternal` will ever 'add' a mirror - and that updates `Names` (so should be covered).  

The handling of `Names` already in `updateImageFromDatabase` (or more specifically `fetchFromDatabase`) will keep the mirror name already in Names.

What's missing here is adding the names from the `existingImage` to this list, if the image is already in central-db this may overwrite whats there with just the mirror and source name.

Reply:
In the future believe we should consider separating 'Names' from `Image` so they can be updated independently.